src/units/unit.cpp

```diff
@@ -2633,6 +2633,29 @@ void unit::clear_changed_attributes()
 	}
 }
 
+std::vector<t_string> unit::unit_special_notes() const {
```

Comment:
It seems this function is never called, the special notes are only accessed by the help_topic_generator, which only uses the functions in units/types.cpp.

Reply:
Hmm… I think there _should_ be some place where this is called though, even if it doesn't exist currently… after all, there's support for an individual unit to have custom special notes.

Reply:
Probably, but as it needs to change in this PR and it's currently an untested code path, I'd prefer to just have a comment saying to consider copying it from types.cpp when someone has a test-case.

Reply:
That would make more sense if not for the fact that it's basically a clone of the equivalent function in `unit_type`.

Reply:
Why would it make sense to have an unused and untested codepath instead of a comment?

Reply:
If it's essentially copy-paste of a used and tested codepath then it's not much different from it itself being used and tested.

Reply:
@irydacea, please can we have your opinion on this? I think it would be better to just delete the unused code.

Reply:
Which function is used for the sidebar description tooltips? Considering that units having different attributes than their unit types is actually an extremely common thing.

Reply:
Thanks for that question, I've opened PR #5808 to include the special notes in the tooltip. Haven't tested it with units that have different attributes to their unit types, but handling that seems more suited to the current PR (and if it needed any changes then it would add a merge conflict with the current PR).

@CelticMinstrel the cut & pasted code in the current PR has a bug that causes duplicated special notes for attacks, an example unit that shows this is the Lancer.

Reply:
Okay, so I'll need to call `std::unique` too…